compiler/rustc_session/src/config.rs

```diff
@@ -1980,6 +1981,34 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
         }
     }
 
+    // Try to find a directory containing the Rust `src`, for more details see
+    // the doc comment on the `real_rust_source_base_dir` field.
+    let tmp_buf;
+    let sysroot = match &sysroot_opt {
+        Some(s) => s,
+        None => {
+            tmp_buf = crate::filesearch::get_or_default_sysroot();
+            &tmp_buf
+        }
+    };
```

Comment:
Computing the sysroot here is a problem as it means when the `config` driver callback changes the sysroot, that does not properly get taken into account.

I'm trying to fix this in https://github.com/rust-lang/rust/pull/122993 but I have no idea about all this "tracked" stuff, so I don't know if what I do makes any sense.